python/lsst/ts/MTMount/mock/controller.py

```diff
@@ -46,26 +62,38 @@ class Controller:
 
     Parameters
     ----------
-    reply_port : `int`
-        Port for writing replies (that the CSC reads).
-        The command port is one greater than the reply port.
+    command_port : `int`
+        Port for reading commands (that the CSC writes).
+        The reply port is one greater than the command port.
     log : `logging.Logger`
         Logger.
+    reconnect : `bool` (optional)
+        Try to reconnect if the connection is lost?
+        Defaults to False for unit tests.
+    telemetry_dict : `dict`
+        Dict of telemetry topic name: data dict
+        where data dict is a dict of key: value
+        for the public fields of a telemetry topic.
     """
 
-    def __init__(self, reply_port, log):
+    def __init__(self, command_port, log, reconnect=False, telemetry_dict=None):
+        self.command_port = command_port
         self.log = log.getChild("Controller")
-        self.communicator = communicator.Communicator(
-            name="Controller",
-            client_host=salobj.LOCAL_HOST,
-            client_port=reply_port,
-            server_host=salobj.LOCAL_HOST,
-            server_port=reply_port + 1,
-            log=self.log,
-            read_replies=False,
-            connect_client=False,
-            connect_callback=self.connect_callback,
-        )
+        self.reconnect = reconnect
+
+        if telemetry_dict is None:
+            import yaml
```

Comment:
isn't it better to have the import at the top of the file.. I'm weary of import statements in the middle of the code... any reason for doing this?

Reply:
That was a temporary hack that I have since removed. I think you reviewed this pull request before it was ready.